beanie/odm/fields.py

```diff
@@ -414,7 +414,7 @@ def __get_pydantic_core_schema__(
                     lambda instance: cls.serialize(instance)  # type: ignore
                 ),
             )
-            return core_schema.general_plain_validator_function(
+            return core_schema.with_info_plain_validator_function(
                 cls.build_validation(handler, source_type)
             )
```

Comment:
I was reading the original code, and my IDE seems to complain that this code is unreachable.

Reply:
Yeah that was also my case but as my PR is only related to deprecation I only changed the deprecated calls